src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,35 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private Map<String, UserData> cache;
```

Comment:
It's a good practice to specify the type of Map in terms of concurrency. Although you've used ConcurrentHashMap, the reference type is just Map. It's more expressive to use ConcurrentMap as the reference type when you're specifically choosing a concurrent implementation for its thread-safety features.